sql/sql_expression_operations.cc

```diff
@@ -2505,14 +2505,22 @@ BoundSqlExpression
 CaseExpression::
 bind(SqlBindingScope & scope) const
 {
+    std::shared_ptr<ExpressionValueInfo> info;
+
     BoundSqlExpression boundElse;
-    if (elseExpr)
+    if (elseExpr) {
         boundElse = elseExpr->bind(scope);
+        info = boundElse.info;
+    }
 
     std::vector<std::pair<BoundSqlExpression, BoundSqlExpression> > boundWhen;
 
     for (auto & w: when) {
         boundWhen.emplace_back(w.first->bind(scope), w.second->bind(scope));
+        if (info)
+            info = std::make_shared<ORExpressionValueInfo>(info, boundWhen.back().second.info);
```

Comment:
I added the minimum and we can add more in a later story. But the very reason we created this info type was to avoid creating and maintaining the whole combinatory tree.